### PR TITLE
Descriptor Cluster Adding TagList attribute

### PR DESCRIPTION
**Pull request by abeck-whirlpool**
Adding Tag List attribute to Descriptor Cluster to match spec
Regen all files

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -169,5 +169,6 @@
     "C_Cpp.default.cppStandard": "gnu++14",
     "C_Cpp.default.cStandard": "gnu11",
     "cmake.configureOnOpen": false,
-    "search.followSymlinks": false
+    "search.followSymlinks": false,
+    "cmake.sourceDirectory": "/Users/austinbeck/Desktop/connectedhomeip/examples/bridge-app/esp32"
```
Comment (bzbarsky-apple) [inline .vscode/settings.json:173]: This is highly unlikely to work for anyone else.  Pleaes do not commit this.

File: src/app/zap-templates/zcl/data-model/chip/descriptor-cluster.xml [hunk 2]
```diff
@@ -23,6 +23,14 @@ limitations under the License.
     <item name="Revision" type="INT16U"/>
   </struct>
 
+   <struct name="SemanticTagStruct">
+    <cluster code="0x001d"/>
+    <item name="MfgCode" type="VENDOR_ID" isNullable="true"/>
+    <item name="Namespace" type="ENUM8"/>
+    <item name= "Tag" type="ENUM8"/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/descriptor-cluster.xml:30]:
```suggestion
    <item name="Tag" type="ENUM8"/>
```

**Review by bzbarsky-apple — APPROVED** [review]
It looks like the "tag" member name on some generated struct is colliding with a function argument name, so we probably need a separate PR to change the codegen template to use a different function argument name...

File: .vscode/settings.json (hunk 1 above)
Comment (abeck-whirlpool) [reply .vscode/settings.json]: Not sure why that occurred but I reverted it/removed that line

**Comment by woody-apple** [comment]
@woody-apple to follow up.

File: src/app/zap-templates/zcl/data-model/chip/descriptor-cluster.xml [hunk 3]
```diff
@@ -33,5 +41,6 @@ limitations under the License.
     <attribute side="server" code="0x0001" define="SERVER_LIST" type="ARRAY" entryType="CLUSTER_ID" writable="false" optional="false">ServerList</attribute>
     <attribute side="server" code="0x0002" define="CLIENT_LIST" type="ARRAY" entryType="CLUSTER_ID" writable="false" optional="false">ClientList</attribute>
     <attribute side="server" code="0x0003" define="PARTS_LIST" type="ARRAY" entryType="ENDPOINT_NO" writable="false" optional="false">PartsList</attribute>
+    <attribute side="server" code="0x0004" define="TAG_LIST" type="ARRAY" entryType="SemanticTagStruct" writable="false" optional="true">TagList</attribute>
```
Comment (cliffamzn) [inline src/app/zap-templates/zcl/data-model/chip/descriptor-cluster.xml:44]: can we add a length=6 tag here too?
```suggestion
    <attribute side="server" code="0x0004" define="TAG_LIST" type="ARRAY" entryType="SemanticTagStruct" writable="false" optional="true" length="6">TagList</attribute>
```

**Review by woody-apple — APPROVED** [review]

**Review by cliffamzn — APPROVED** [review]

**Pull request merged**